doc/source/data/aggregations.rst

```diff
@@ -0,0 +1,148 @@
+
+    import ray
+    from ray.data.aggregate import Count, Mean, Min, Max, Std
+
+    ds = ray.data.range(100)
+    ds = ds.add_column("group_key", lambda x: x % 3)
+
+    # Compute multiple aggregations at once
+    result = ds.groupby("group_key").aggregate(
+        Count(on="id"),
+        Mean(on="id"),
+        Min(on="id"),
+        Max(on="id"),
+        Std(on="id")
+    ).take_all()
+    # result: [{'group_key': 0, 'count(id)': 34, 'mean(id)': ..., 'min(id)': ..., 'max(id)': ..., 'std(id)': ...},
+    #          {'group_key': 1, 'count(id)': 33, 'mean(id)': ..., 'min(id)': ..., 'max(id)': ..., 'std(id)': ...},
+    #          {'group_key': 2, 'count(id)': 33, 'mean(id)': ..., 'min(id)': ..., 'max(id)': ..., 'std(id)': ...}]
+
+
+Custom Aggregations
+------------------
+
+For more complex aggregation needs, Ray Data allows you to create custom aggregations by implementing the :class:`~ray.data.aggregate.AggregateFnV2` interface. The AggregateFnV2 interface provides a framework for implementing distributed aggregations with three key methods:
+
+1. `aggregate_block`: Processes a single block of data and returns a partial aggregation result
+2. `combine`: Merges two partial aggregation results into a single result
+3. `_finalize`: Transforms the final accumulated result into the desired output format
+
+The aggregation process follows these steps:
+
+1. **Initialization**: For each group (if grouping) or for the entire dataset, an initial accumulator is created using `zero_factory`
+2. **Block Aggregation**: The `aggregate_block` method is applied to each block independently
+3. **Combination**: The `combine` method merges partial results into a single accumulator
+4. **Finalization**: The `_finalize` method transforms the final accumulator into the desired output
+
+Example: Creating a Missing Value Percentage Aggregator
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Here's an example of creating a custom aggregator that calculates the percentage of null values in a column:
+
+.. testcode::
+
+    from ray.data.aggregate import AggregateFnV2
+    from ray.data.block import Block, BlockAccessor
+    import pyarrow.compute as pc
+    from typing import List, Optional
+
+    class MissingValuePercentage(AggregateFnV2):
+        def __init__(self, on: str):
+            # Initialize with a list accumulator [null_count, total_count]
+            super().__init__(
+                f"missing_pct({on})",
+                on=on,
+                ignore_nulls=False,
+                zero_factory=lambda: [0, 0]
+            )
+
+        def aggregate_block(self, block: Block) -> List[int]:
+            # Process a single block of data
+            block_acc = BlockAccessor.for_block(block)
+            table = block_acc.to_arrow()
+            column = table.column(self._target_col_name)
+            
+            total_count = len(column)
+            null_count = pc.sum(pc.is_null(column, nan_is_null=True).cast("int32")).as_py()
```

Comment:
Let's avoid giving examples with conversion, instead just use BlockColumnAccessor so that it works for both Pandas/PA